adcm_client/objects.py

```diff
@@ -296,29 +272,30 @@ def config(self, full=False):
 
     @allure_step("Save config")
     def config_set(self, data):
+        config = self.config(full=True)
         if "config" in data and "attr" in data:
             # We are in a new mode with full_info == True
             if data["attr"] is None:
-                data["attr"] = []
-            history_entry = self._subcall(
-                'config', 'history', 'create',
-                config=data["config"],
-                attr=data["attr"],
-            )
+                data["attr"] = {}
+            history_entry = self._subcall('config', 'history', 'create', **data)
             return history_entry
-        history_entry = self._subcall('config', 'history', 'create', config=data)
+        history_entry = self._subcall(
+            'config', 'history', 'create', config=data, attr=config['attr'])
         return history_entry['config']
 
     @allure_step("Save config")
-    def config_set_diff(self, data):
-        config = self.config()
-        for gk, gv in data.items():
-            if isinstance(gv, dict):
-                for k, v in gv.items():
-                    config[gk][k] = v
-            else:
-                config[gk] = gv
-        self.config_set(config)
+    def config_set_diff(self, data, full=False):
```

Comment:
Why do you need to pass full explicit? You could find out about from data structure as your done in config config_set function.

Reply:
I find it a bad idea to explicitly check for the presence of keys in a structure